src/actions-on-google.ts

```diff
@@ -509,7 +509,7 @@ export class ActionsOnGoogle {
                 if (data.dialog_state_out) {
                     this._conversationState = data.dialog_state_out.conversation_state
                     if (data.dialog_state_out.supplemental_display_text &&
-                        !assistResponse.displayText) {
+                        !assistResponse.displayText.length) {
```

Comment:
Maybe we should add a test for this for the future?

Reply:
Yeah, always a good idea. I will add one.

Reply:
I finally had some time to take a look at this again. Unfortunately, I don't see a good way to write a test for this the way things are structured. The error is in the conversion of the `AssistantSdkResponse` to an `AssistantResponse`. All the existing tests already assume we have an `AssistantResponse`. To test this conversion, I believe we would have to refactor the logic in

```typescript
conversation.on('data', (data: AssistantSdkResponse) => {
...
}
```

into a separate function. Then I could write a test for just that function. But that's a fair amount of refactoring. I wanted to check with you first to make sure there wasn't an easier way.

Reply:
I've been thinking about this for a while, and I agree right now the tests aren't set up for this. I guess it's fine to omit tests for the change.